File: ios/Talise/Features/Chat/AgentIntentCard.swift
import SwiftUI

/// Renders a parsed Talise Agent intent beneath the assistant's message.
///
///   • Read-only intents (balance / yield / activity) auto-run inline on
///     appear and show their results — no slide, no signing.
///   • Write intents call `POST /api/agent/plan` to validate + price, render a
///     per-step preview, and gate execution behind simple Accept / Decline
///     buttons — enabled only when the server says the plan is `confirmable`.
///
/// "Agent proposes → server validates → human confirms." Styling mirrors
/// `PayTeamView` (rampCard, mint accept button, honest error copy).
struct AgentIntentCard: View {
    let intent: AgentIntent

    @Environment(AppSession.self) private var session

    private enum Stage { case loading, plan, running, done, readOnly, failed, declined }

    @State private var stage: Stage = .loading
    @State private var plan: AgentPlanDTO?
    @State private var resultLines: [String] = []
    @State private var error: String?

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            switch stage {
            case .loading:
                loadingRow
            case .readOnly:
                readOnlyBody
            case .plan, .running, .failed:
                planBody
            case .done:
                doneBody
            case .declined:
                declinedBody
            }
        }
        .padding(16)
        .rampCard()
        .task { await start() }
    }

    // MARK: - Stages

    private var loadingRow: some View {
        HStack(spacing: 8) {
            ProgressView().controlSize(.mini).tint(TaliseColor.fgDim)
            Text(intent.isReadOnlyOnly ? "Looking that up…" : "Checking this plan…")
                .font(TaliseFont.body(13, weight: .light))
                .foregroundStyle(TaliseColor.fgMuted)
        }
    }

    private var readOnlyBody: some View {
        VStack(alignment: .leading, spacing: 8) {
            ForEach(Array(resultLines.enumerated()), id: \.offset) { _, line in
                HStack(alignment: .top, spacing: 8) {
                    Circle().fill(TaliseColor.fgDim).frame(width: 5, height: 5).padding(.top, 6)
                    Text(line)
                        .font(TaliseFont.body(14, weight: .regular))
                        .foregroundStyle(TaliseColor.fg)
                        .fixedSize(horizontal: false, vertical: true)
                }
            }
            if resultLines.isEmpty {
                Text("Nothing to show.")
                    .font(TaliseFont.body(13, weight: .light))
                    .foregroundStyle(TaliseColor.fgMuted)
            }
        }
    }

    @ViewBuilder
    private var planBody: some View {
        if let plan {
            // Summary header
            Text(plan.summary)
                .font(TaliseFont.body(14, weight: .medium))
                .foregroundStyle(TaliseColor.fg)
                .fixedSize(horizontal: false, vertical: true)

            VStack(alignment: .leading, spacing: 10) {
                ForEach(Array(plan.steps.enumerated()), id: \.offset) { _, step in
                    stepRow(step)
                }
            }

            if let limit = plan.limit {
                Text("\(limit.window.capitalized) limit \(TaliseFormat.usd2(limit.limit)) · used \(TaliseFormat.usd2(limit.used)).")
                    .font(TaliseFont.mono(10, weight: .light)).kerning(0.2)
                    .foregroundStyle(TaliseColor.fgDim)
                    .fixedSize(horizontal: false, vertical: true)
            }

            if let error {
                Text(error)
                    .font(TaliseFont.body(13, weight: .light))
                    .foregroundStyle(TaliseColor.danger)
                    .fixedSize(horizontal: false, vertical: true)
            }

            if plan.confirmable {
                confirmButtons(plan)
                    .padding(.top, 2)
                gaslessNote
            }
        }
    }

    /// Accept / Decline — simple buttons (the user asked for these over a slide).
    /// Accept runs the validated plan; Decline dismisses it without moving money.
    private func confirmButtons(_ plan: AgentPlanDTO) -> some View {
        HStack(spacing: 10) {
            Button { decline() } label: {
                Text("Decline")
                    .font(TaliseFont.body(15, weight: .semibold))
                    .foregroundStyle(TaliseColor.fgMuted)
                    .frame(maxWidth: .infinity).frame(height: 48)
                    .background(Capsule().fill(TaliseColor.surface2))
            }
            .buttonStyle(.plain)
            .disabled(stage == .running)

            Button { Task { await confirm() } } label: {
                Group {
                    if stage == .running {
                        HStack(spacing: 7) {
                            ProgressView().controlSize(.mini).tint(.black)
                            Text("Working…")
                        }
                    } else {
                        Text(acceptTitle(plan))
                    }
                }
                .font(TaliseFont.body(15, weight: .semibold))
                .foregroundStyle(.black)
                .frame(maxWidth: .infinity).frame(height: 48)
                .background(Capsule().fill(TaliseColor.greenMint))
                .opacity(stage == .running ? 0.7 : 1)
            }
            .buttonStyle(.plain)
            .disabled(stage == .running)
        }
    }

    private var declinedBody: some View {
        Text("Okay, I didn't run that. Tell me what to change.")
            .font(TaliseFont.body(14, weight: .regular))
            .foregroundStyle(TaliseColor.fgMuted)
            .fixedSize(horizontal: false, vertical: true)
    }

    private var doneBody: some View {
        VStack(alignment: .leading, spacing: 10) {
            HStack(spacing: 8) {
                Image(systemName: "checkmark.seal.fill")
                    .font(.system(size: 18)).foregroundStyle(TaliseColor.greenMint)
                Text("Done")
                    .font(TaliseFont.heading(16, weight: .medium))
                    .foregroundStyle(TaliseColor.fg)
            }
            ForEach(Array(resultLines.enumerated()), id: \.offset) { _, line in
                Text(line)
                    .font(TaliseFont.body(14, weight: .regular))
                    .foregroundStyle(TaliseColor.fgMuted)
                    .fixedSize(horizontal: false, vertical: true)
            }
        }
    }

    // MARK: - Step row

    private func stepRow(_ step: PlannedStepDTO) -> some View {
        HStack(alignment: .top, spacing: 10) {
            statusIcon(step)
            VStack(alignment: .leading, spacing: 2) {
                Text(step.label)
                    .font(TaliseFont.body(14, weight: .regular))
                    .foregroundStyle(step.isReadOnly ? TaliseColor.fgMuted : TaliseColor.fg)
                    .fixedSize(horizontal: false, vertical: true)
                if let detail = step.detail, !detail.isEmpty {
                    Text(detail)
                        .font(TaliseFont.body(12, weight: .light))
                        .foregroundStyle(step.isBlocked ? TaliseColor.danger : TaliseColor.fgDim)
                        .fixedSize(horizontal: false, vertical: true)
                }
            }
            Spacer(minLength: 0)
        }
    }

    @ViewBuilder
    private func statusIcon(_ step: PlannedStepDTO) -> some View {
        if step.isBlocked {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 13)).foregroundStyle(TaliseColor.danger).padding(.top, 1)
        } else if step.isReadOnly {
            Image(systemName: "eye")
                .font(.system(size: 12)).foregroundStyle(TaliseColor.fgDim).padding(.top, 2)
        } else {
            Image(systemName: "checkmark.circle.fill")
                .font(.system(size: 13)).foregroundStyle(TaliseColor.greenMint).padding(.top, 1)
        }
    }

    private var gaslessNote: some View {
        HStack(spacing: 7) {
            Image(systemName: "bolt.fill")
                .font(.system(size: 10, weight: .medium)).foregroundStyle(TaliseColor.greenMint)
            Text("No network fee. Talise sponsors the gas.")
                .font(TaliseFont.mono(10, weight: .light)).kerning(0.2)
                .foregroundStyle(TaliseColor.fgDim)
        }
        .frame(maxWidth: .infinity, alignment: .center)
    }

    private func acceptTitle(_ plan: AgentPlanDTO) -> String {
        plan.totalSendUsd > 0
            ? "Accept · \(TaliseFormat.usd2(plan.totalSendUsd))"
            : "Accept"
    }

    private func decline() {
        guard stage != .running else { return }
        error = nil
        stage = .declined
    }

    // MARK: - Actions

    private func start() async {
        if intent.isReadOnlyOnly {
            do {
                resultLines = try await AgentExecutor.runReadOnly(intent.steps)
                stage = .readOnly
            } catch {
                if APIError.isCancellation(error) { return }
                self.error = APIError.honestMoneyError(error, fallback: "Couldn't load that right now.")
                stage = .failed
            }
        } else {
            do {
                plan = try await AgentPlanAPI.plan(steps: intent.steps)
                stage = .plan
            } catch {
                if APIError.isCancellation(error) { return }
                self.error = APIError.honestMoneyError(error, fallback: "Couldn't check that plan right now.")
                // Surface as a one-line failed plan card.
                plan = AgentPlanDTO(confirmable: false, steps: [], totalSendUsd: 0, limit: nil,
                                    summary: "Couldn't check this plan.")
                stage = .failed
            }
        }
    }

    private func confirm() async {
        guard let plan, stage != .running else { return }
        stage = .running
        error = nil
        do {
            resultLines = try await AgentExecutor.execute(plan: plan, intent: intent)
            stage = .done
        } catch ZkLoginCoordinator.SessionError.rebindRequired {
            error = "Sign in again. Your session needs a refresh."
            session.signOut()
            stage = .failed
        } catch {
            if APIError.isCancellation(error) { return }
            self.error = APIError.honestMoneyError(error, fallback: "Couldn't complete that. Please try again.")
            // Keep the plan visible so the user can tap Accept to retry.
            stage = .plan
        }
    }
}

File: android/app/src/main/java/io/talise/app/feature/chat/AgentIntentCard.kt
package io.talise.app.feature.chat

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Bolt
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Link
import androidx.compose.material.icons.filled.Share
import androidx.compose.material.icons.filled.Verified
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material.icons.outlined.Visibility
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import io.talise.app.ui.components.rampCard
import io.talise.app.ui.theme.TaliseColors
import io.talise.app.ui.theme.TaliseType
import java.util.Locale

/**
 * Renders a parsed Talise Agent intent beneath the assistant's message — the
 * Android port of iOS `AgentIntentCard`.
 *
 *   • Read-only intents (balance / yield / activity) auto-run inline on
 *     appear and show their results — no confirm, no signing.
 *   • Write intents call `POST /api/agent/plan` to validate + price, render a
 *     per-step preview, and gate execution behind simple Accept / Decline
 *     buttons — enabled only when the server says the plan is `confirmable`.
 *
 * "Agent proposes, server validates, human confirms."
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AgentIntentCard(
    intent: AgentIntent,
    executed: List<AgentActionResult>? = null,
    /**
     * Runs the validated plan on a scope that survives this composable
     * ([ChatViewModel.executeIntent]) — the card can scroll out of the
     * LazyColumn mid-transfer, and money movement must not die with it.
     */
    executePlan: (AgentPlanDTO, AgentIntent, onResult: (List<AgentActionResult>?, String?) -> Unit) -> Unit,
) {
    var stage by remember { mutableStateOf(Stage.Loading) }
    var plan by remember { mutableStateOf<AgentPlanDTO?>(null) }
    var resultLines by remember { mutableStateOf<List<String>>(emptyList()) }
    var actionResults by remember { mutableStateOf<List<AgentActionResult>>(emptyList()) }
    var receiptFor by remember { mutableStateOf<AgentActionResult?>(null) }
    var error by remember { mutableStateOf<String?>(null) }

    LaunchedEffect(Unit) {
        // Already ran in a prior session — open straight to the receipt and
        // never re-fetch a plan or re-prompt for a transfer that's done.
        if (!executed.isNullOrEmpty()) {
            actionResults = executed
            stage = Stage.Done
            return@LaunchedEffect
        }
        if (intent.isReadOnlyOnly) {
            try {
                resultLines = AgentExecutor.runReadOnly(intent.steps)
                stage = Stage.ReadOnly
            } catch (t: Throwable) {
                error = t.message ?: "Couldn't load that right now."
                stage = Stage.Failed
            }
        } else {
            try {
                plan = AgentApi.instance.plan(AgentPlanRequest(steps = intent.steps))
                stage = Stage.Plan
            } catch (t: Throwable) {
                error = t.message ?: "Couldn't check that plan right now."
                // Surface as a one-line failed plan card.
                plan = AgentPlanDTO(confirmable = false, steps = emptyList(), totalSendUsd = 0.0, limit = null, summary = "Couldn't check this plan.")
                stage = Stage.Failed
            }
        }
    }

    fun decline() {
        if (stage == Stage.Running) return
        error = null
        stage = Stage.Declined
    }

    fun confirm() {
        val p = plan ?: return
        if (stage == Stage.Running) return
        stage = Stage.Running
        error = null
        // Persisting the outcome on the transcript happens inside the runner
        // (ChatViewModel), so a reopen shows the receipt even if this card is
        // gone by the time the transfer completes.
        executePlan(p, intent) { results, failure ->
            if (results != null) {
                actionResults = results
                stage = Stage.Done
            } else {
                error = failure ?: "Couldn't complete that. Please try again."
                // Keep the plan visible so the user can tap Accept to retry.
                stage = Stage.Plan
            }
        }
    }

    Column(
        Modifier
            .fillMaxWidth()
            .rampCard()
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        when (stage) {
            Stage.Loading -> LoadingRow(intent)
            Stage.ReadOnly -> ReadOnlyBody(resultLines)
            Stage.Plan, Stage.Running, Stage.Failed -> PlanBody(
                plan = plan,
                error = error,
                running = stage == Stage.Running,
                onDecline = { decline() },
                onConfirm = { confirm() },
            )
            Stage.Done -> DoneBody(actionResults, onReceipt = { receiptFor = it })
            Stage.Declined -> Text(
                "Okay, I didn't run that. Tell me what to change.",
                style = TaliseType.body(14.sp),
                color = TaliseColors.fgMuted,
            )
        }
    }

    receiptFor?.let { r ->
        ModalBottomSheet(
            onDismissRequest = { receiptFor = null },
            sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true),
            containerColor = TaliseColors.bg,
        ) {
            AgentReceiptSheet(
                amountUsd = r.amountUsd ?: 0.0,
                recipient = r.recipient.orEmpty(),
                digest = r.digest.orEmpty(),
                title = receiptTitle(r.kind),
            )
        }
    }
}

private enum class Stage { Loading, Plan, Running, Done, ReadOnly, Failed, Declined }

@Composable
private fun LoadingRow(intent: AgentIntent) {
    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
        CircularProgressIndicator(
            modifier = Modifier.size(14.dp),
            color = TaliseColors.fgDim,
            strokeWidth = 1.5.dp,
        )
        Text(
            if (intent.isReadOnlyOnly) "Looking that up…" else "Checking this plan…",
            style = TaliseType.body(13.sp, FontWeight.Light),
            color = TaliseColors.fgMuted,
        )
    }
}

@Composable
private fun ReadOnlyBody(resultLines: List<String>) {
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        resultLines.forEach { line ->
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                Box(
                    Modifier
                        .padding(top = 6.dp)
                        .size(5.dp)
                        .background(TaliseColors.fgDim, CircleShape),
                )
                Text(line, style = TaliseType.body(14.sp), color = TaliseColors.fg)
            }
        }
        if (resultLines.isEmpty()) {
            Text("Nothing to show.", style = TaliseType.body(13.sp, FontWeight.Light), color = TaliseColors.fgMuted)
        }
    }
}

@Composable
private fun PlanBody(
    plan: AgentPlanDTO?,
    error: String?,
    running: Boolean,
    onDecline: () -> Unit,
    onConfirm: () -> Unit,
) {
    if (plan == null) {
        // A failed read-only run lands here with no plan — show the honest error.
        if (error != null) {
            Text(error, style = TaliseType.body(13.sp, FontWeight.Light), color = TaliseColors.danger)
        }
        return
    }
    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
        // Summary header
        Text(plan.summary, style = TaliseType.body(14.sp, FontWeight.Medium), color = TaliseColors.fg)

        Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
            plan.steps.forEach { step -> StepRow(step) }
        }

        plan.limit?.let { limit ->
            val window = limit.window.replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.US) else it.toString() }
            Text(
                "$window limit ${AgentExecutor.usd2(limit.limit)} · used ${AgentExecutor.usd2(limit.used)}.",
                style = TaliseType.mono(10.sp, FontWeight.Light).copy(letterSpacing = 0.2.sp),
                color = TaliseColors.fgDim,
            )
        }

        if (error != null) {
            Text(error, style = TaliseType.body(13.sp, FontWeight.Light), color = TaliseColors.danger)
        }

        if (plan.confirmable) {
            ConfirmButtons(plan, running = running, onDecline = onDecline, onConfirm = onConfirm)
            GaslessNote()
        }
    }
}

/**
 * Accept / Decline — simple buttons. Accept runs the validated plan; Decline
 * dismisses it without moving money.
 */
@Composable
private fun ConfirmButtons(
    plan: AgentPlanDTO,
    running: Boolean,
    onDecline: () -> Unit,
    onConfirm: () -> Unit,
) {
    Row(
        Modifier.fillMaxWidth().padding(top = 2.dp),
        horizontalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Box(
            Modifier
                .weight(1f)
                .height(48.dp)
                .clip(RoundedCornerShape(24.dp))
                .background(TaliseColors.surface2)
                .clickable(enabled = !running) { onDecline() },
            contentAlignment = Alignment.Center,
        ) {
            Text("Decline", style = TaliseType.body(15.sp, FontWeight.SemiBold), color = TaliseColors.fgMuted)
        }

        Box(
            Modifier
                .weight(1f)
                .height(48.dp)
                .clip(RoundedCornerShape(24.dp))
                .background(TaliseColors.greenMint)
                .alpha(if (running) 0.7f else 1f)
                .clickable(enabled = !running) { onConfirm() },
            contentAlignment = Alignment.Center,
        ) {
            if (running) {
                Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(7.dp)) {
                    CircularProgressIndicator(
                        modifier = Modifier.size(14.dp),
                        color = TaliseColors.bg,
                        strokeWidth = 1.5.dp,
                    )
                    Text("Working…", style = TaliseType.body(15.sp, FontWeight.SemiBold), color = TaliseColors.bg)
                }
            } else {
                val title = if (plan.totalSendUsd > 0) "Accept · ${AgentExecutor.usd2(plan.totalSendUsd)}" else "Accept"
                Text(title, style = TaliseType.body(15.sp, FontWeight.SemiBold), color = TaliseColors.bg)
            }
        }
    }
}

@Composable
private fun StepRow(step: PlannedStepDTO) {
    Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
        when {
            step.isBlocked -> Icon(
                Icons.Filled.Warning, contentDescription = null,
                tint = TaliseColors.danger, modifier = Modifier.padding(top = 1.dp).size(15.dp),
            )
            step.isReadOnly -> Icon(
                Icons.Outlined.Visibility, contentDescription = null,
                tint = TaliseColors.fgDim, modifier = Modifier.padding(top = 2.dp).size(14.dp),
            )
            else -> Icon(
                Icons.Filled.CheckCircle, contentDescription = null,
                tint = TaliseColors.greenMint, modifier = Modifier.padding(top = 1.dp).size(15.dp),
            )
        }
        Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(
                step.label,
                style = TaliseType.body(14.sp),
                color = if (step.isReadOnly) TaliseColors.fgMuted else TaliseColors.fg,
            )
            val detail = step.detail
            if (!detail.isNullOrEmpty()) {
                Text(
                    detail,
                    style = TaliseType.body(12.sp, FontWeight.Light),
                    color = if (step.isBlocked) TaliseColors.danger else TaliseColors.fgDim,
                )
            }
        }
    }
}

@Composable
private fun GaslessNote() {
    Row(
        Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(Icons.Filled.Bolt, contentDescription = null, tint = TaliseColors.greenMint, modifier = Modifier.size(12.dp))
        Spacer(Modifier.width(7.dp))
        Text(
            "No network fee. Talise sponsors the gas.",
            style = TaliseType.mono(10.sp, FontWeight.Light).copy(letterSpacing = 0.2.sp),
            color = TaliseColors.fgDim,
        )
    }
}

@Composable
private fun DoneBody(actionResults: List<AgentActionResult>, onReceipt: (AgentActionResult) -> Unit) {
    val context = androidx.compose.ui.platform.LocalContext.current
    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            Icon(Icons.Filled.Verified, contentDescription = null, tint = TaliseColors.greenMint, modifier = Modifier.size(18.dp))
            Text("Done", style = TaliseType.heading(16.sp, FontWeight.Medium), color = TaliseColors.fg)
        }
        actionResults.forEach { r ->
            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Text(r.line, style = TaliseType.body(14.sp), color = TaliseColors.fgMuted)
                // A confirmed on-chain money step gets a "Share receipt" chip.
                if (r.digest != null && (r.amountUsd ?: 0.0) > 0) {
                    ResultChip(icon = Icons.Filled.Share, label = "Share receipt") { onReceipt(r) }
                }
                // A payment-link (request) step gets a "Share link" chip.
                r.link?.let { link ->
                    ResultChip(icon = Icons.Filled.Link, label = "Share link") {
                        val share = android.content.Intent(android.content.Intent.ACTION_SEND).apply {
                            type = "text/plain"
                            putExtra(android.content.Intent.EXTRA_TEXT, link)
                        }
                        context.startActivity(android.content.Intent.createChooser(share, "Share link"))
                    }
                }
            }
        }
    }
}

@Composable
private fun ResultChip(icon: androidx.compose.ui.graphics.vector.ImageVector, label: String, onClick: () -> Unit) {
    Row(
        Modifier
            .clip(RoundedCornerShape(20.dp))
            .background(TaliseColors.greenMint)
            .clickable { onClick() }
            .padding(horizontal = 14.dp, vertical = 9.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(7.dp),
    ) {
        Icon(icon, contentDescription = null, tint = TaliseColors.bg, modifier = Modifier.size(14.dp))
        Text(label, style = TaliseType.body(13.sp, FontWeight.SemiBold), color = TaliseColors.bg)
    }
}

private fun receiptTitle(kind: String): String = when (kind) {
    "send" -> "Sent"
    "save" -> "Saved"
    "withdraw" -> "Withdrew"
    "claim_rewards" -> "Claimed"
    else -> "Done"
}
